# Représentation d'un texte en machine

https://pixees.fr/informatiquelycee/n_site/nsi_prem_texte.html

### ASCII - À faire vous-même 1

In [ ]:
"a" est 97 en décimal

donc le code binaire du "a" minuscule en ASCII est = 1100001

In [5]:
# trouver en décimal 
ord("a")

97

In [6]:
# trouver la lettre à partir de la donnée décimale 
chr(97)

'a'

### UNICODE - À faire vous-même 2

In [7]:
# trouver en décimal 
ord("b")

98

In [120]:
# convertir en binaire fonction recursive

def convertir_base_2(decimal,liste_reste=[]):
    
    quotient=decimal//2
            
    if quotient != 0:
        
        reste=decimal%2
        decimal=quotient
        liste_reste.append(str(reste))
        convertir_base_2(decimal,liste_reste)
    else:    
        reste=decimal%2
        liste_reste.append(str(reste))
        liste_reste.reverse()
        return liste_reste
                    
    return liste_reste


In [121]:
bin=convertir_base_2(98,[])
"".join(bin)

'1100010'

code binaire du "b" minuscule Unicode codé avec UTF-8 est : 1100010